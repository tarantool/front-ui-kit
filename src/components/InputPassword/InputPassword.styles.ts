import { css } from '@emotion/css';

export const styles = {
  innerButton: css`
    padding: 0;
  `,
  icon: css`
    display: block;
  `,
};
