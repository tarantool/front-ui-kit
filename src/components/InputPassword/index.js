// @flow
import * as React from 'react';
import { css } from '@emotion/css';
import { IconEyeClosed, IconEyeOpened } from '../Icon';
import { Input, type InputProps } from '../Input';
import { Button } from '../Button';

const styles = {
  innerButton: css`
    padding: 0;
  `,
  icon: css`
    display: block;
  `,
};

type InputPasswordState = {
  hidden: boolean,
};

export class InputPassword extends React.Component<InputProps, InputPasswordState> {
  state = { hidden: true };

  toggleState = () => this.setState(({ hidden }) => ({ hidden: !hidden }));

  render() {
    const { hidden } = this.state;
    const Icon = hidden ? IconEyeOpened : IconEyeClosed;
    return (
      <Input
        {...this.props}
        type={hidden ? 'password' : 'text'}
        rightIcon={
          <Button className={styles.innerButton} size="xs" intent="plain" onClick={this.toggleState}>
            <Icon className={styles.icon} />
          </Button>
        }
      />
    );
  }
}
