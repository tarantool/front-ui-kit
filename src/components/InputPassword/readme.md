```js
import { css } from 'emotion';
import { Input } from '../Input';

initialState = { value: 'Value' };

handleChange = e => setState({ value: e.target.value });

<div className={css`display: flex;`}>
  <InputPassword
    value={state.value}
    placeholder='Placeholder'
    onChange={handleChange}
    title='InputPassword'
  />
  <InputPassword
    value={state.value}
    placeholder='Placeholder'
    onChange={handleChange}
    title='InputPassword'
    disabled
  />
</div>
```
