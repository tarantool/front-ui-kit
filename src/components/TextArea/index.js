// @flow
import * as React from 'react';
import { createRef } from 'react';
import { cx } from 'react-emotion';
import { commonInputStyles, commonInputSizes } from '../Input/commonStyles';

export type TextAreaProps = {
  autoComplete?: 'on' | 'off',  // ?
  autoFocus?: boolean,
  className?: string,
  disabled?: boolean,
  error?: boolean,
  name?: string,
  onBlur?: (e: MouseEvent) => any,
  onChange?: (e: SyntheticInputEvent<HTMLTextAreaElement>) => any,
  onFocus?: (e: MouseEvent) => any,
  onKeyDown?: (e: KeyboardEvent) => any,
  onKeyDownCapture?: (e: KeyboardEvent) => any,
  onKeyPress?: (e: KeyboardEvent) => any,
  onKeyPressCapture?: (e: KeyboardEvent) => any,
  onKeyUp?: (e: KeyboardEvent) => any,
  onKeyUpCapture?: (e: KeyboardEvent) => any,
  rows?: number,
  readOnly?: boolean,
  size?: 'm' | 'l',
  title?: string,
  value?: string,
  placeholder?: string
};

type TextAreaState = {
  focused: boolean;
};

export class TextArea extends React.Component<TextAreaProps, TextAreaState> {
  elementRef = createRef<HTMLTextAreaElement>();

  state = {
    focused: false
  };

  focus = () => {
    if (this.elementRef && this.elementRef.current) {
      this.elementRef.current.focus();
    }
  }

  render() {
    const {
      autoComplete,
      autoFocus,
      className,
      onKeyDown,
      onKeyDownCapture,
      onKeyPress,
      onKeyPressCapture,
      onKeyUp,
      onKeyUpCapture,
      disabled,
      error,
      name,
      onChange,
      readOnly,
      rows,
      title,
      value,
      placeholder,
      size,
      ...props
    } = this.props;

    const { focused } = this.state;

    return (
      <div
        className={cx(
          commonInputStyles.outer,
          {
            [commonInputStyles.disabled]: disabled,
            [commonInputStyles.focused]: focused,
            [commonInputStyles.error]: error
          },
          className
        )}
        title={title}
      >
        <textarea
          {...props}
          autoFocus={autoFocus}
          autoComplete={autoComplete}
          className={cx(commonInputStyles.input, commonInputSizes[size || 'l'])}
          disabled={disabled}
          name={name}
          onChange={onChange}
          onKeyDown={onKeyDown}
          onKeyDownCapture={onKeyDownCapture}
          onKeyPress={onKeyPress}
          onKeyPressCapture={onKeyPressCapture}
          onKeyUp={onKeyUp}
          onKeyUpCapture={onKeyUpCapture}
          onBlur={this.handleInputBlur}
          onFocus={this.handleInputFocus}
          value={value}
          placeholder={placeholder}
          rows={rows}
          readOnly={readOnly}
          ref={this.elementRef}
        />
      </div>
    );
  }

  handleInputFocus = () => this.setState({ focused: true });

  handleInputBlur = (e: Object) => {
    this.setState({ focused: false });
    this.props.onBlur && this.props.onBlur(e);
  };
}
