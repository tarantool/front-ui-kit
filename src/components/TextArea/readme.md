```jsx
import { css } from 'emotion';

const wrap = css`
  margin: 16px;
`;

initialState = { value: '' };

updateState = e => setState({ value: e.target.value });

<>
  <TextArea placeholder='Type here' className={wrap} onChange={updateState} value={state.value} />
  <TextArea placeholder='Type here' className={wrap} onChange={updateState} value={state.value} disabled />
  <TextArea placeholder='Type here' className={wrap} onChange={updateState} value={state.value} error />
  <TextArea placeholder='Type here' className={wrap} onChange={updateState} value={state.value} rows={5} />
</>
```
