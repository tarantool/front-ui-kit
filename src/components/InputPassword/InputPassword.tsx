import React, { FC, useState } from 'react';

import { Button } from '../Button';
import { IconEyeClosed, IconEyeOpened } from '../Icon';
import { Input, InputProps } from '../Input';

import { styles } from './InputPassword.styles';

export const InputPassword: FC<InputProps> = (props: InputProps) => {
  const [hidden, setHidden] = useState(false);

  const toggleType = () => setHidden(!hidden);

  const Icon = hidden ? IconEyeOpened : IconEyeClosed;

  return (
    <Input
      type={hidden ? 'text' : 'password'}
      rightIcon={
        <Button className={styles.innerButton} size="xs" intent="plain" onClick={toggleType}>
          <Icon className={styles.icon} />
        </Button>
      }
      {...props}
    />
  );
};
