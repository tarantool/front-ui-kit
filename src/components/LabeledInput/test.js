import React from 'react';
import renderer from 'react-test-renderer';
import { LabeledInput } from './index';
import { InputPassword } from '../InputPassword';
import { TextArea } from '../TextArea';

const testValue = 'qwertyuiop';

const handleChange = () => void 0;

it('LabeledInput renders correctly', () => {
  const tree = renderer.create(
    <>
      <LabeledInput
        label='Label'
        id='1'
        info='Info LabeledInput component'
        message='Field is required'
        value={testValue}
        placeholder='Placeholder'
        onChange={handleChange}
        title='Input'
        error
      />
      <LabeledInput
        inputComponent={InputPassword}
        label='Label'
        id='2'
        subTitle='Sub title'
        value={testValue}
        placeholder='Placeholder'
        onChange={handleChange}
        title='Input'
      />
      <LabeledInput
        label='Label'
        id='3'
        subTitle='Sub title'
        message='Field is required'
        value={testValue}
        placeholder='Placeholder'
        onChange={handleChange}
        title='Input'
      />
      <LabeledInput
        label='Multiline labeled input'
        id='4'
        subTitle='Sub title'
        message='Field is required'
        inputComponent={TextArea}
        value={testValue}
        rows={4}
        placeholder='Placeholder'
        onChange={handleChange}
        title='Input'
      />
    </>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
