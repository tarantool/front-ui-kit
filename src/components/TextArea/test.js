import React from 'react';
import renderer from 'react-test-renderer';
import { TextArea } from '../TextArea';

const updateState = jest.fn();

it('TextArea renders correctly', () => {
  const tree = renderer.create(
    <>
      <TextArea onChange={updateState} value='Example text' />
      <TextArea onChange={updateState} value='Example text' error />
      <TextArea onChange={updateState} value='Example text' disabled />
      <TextArea onChange={updateState} value='Example text' rows={5} />

      <TextArea onChange={updateState} size='m' value='Example text' />
      <TextArea onChange={updateState} size='m' value='Example text' error />
      <TextArea onChange={updateState} size='m' value='Example text' disabled />
      <TextArea onChange={updateState} size='m' value='Example text' rows={5} />
    </>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
