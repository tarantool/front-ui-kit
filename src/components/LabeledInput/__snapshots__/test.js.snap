// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabeledInput renders correctly 1`] = `
Array [
  <label
    className="css-kf3i1p"
  >
    <div
      className="css-13kfgn1"
    >
      <span
        className="css-dpor9z"
      >
        Label
        <div
          className="css-vj79ep"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onScroll={[Function]}
          placement="top"
        >
          <svg
            className="css-n25ly7"
            viewBox="0 0 12 12"
          >
            <use
              xlinkHref="#info"
            />
          </svg>
        </div>
      </span>
    </div>
    <div
      className="css-w03906"
      title="Input"
    >
      <input
        className="css-1ssefzq"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Placeholder"
        value="qwertyuiop"
      />
    </div>
    <p
      className="css-1ddecx1"
    >
      Field is required
    </p>
  </label>,
  <label
    className="css-kf3i1p"
  >
    <div
      className="css-13kfgn1"
    >
      <span
        className="css-dpor9z"
      >
        Label
      </span>
      <span
        className="css-1dhlryp"
      >
        Sub title
      </span>
    </div>
    <div
      className="css-1prx3ty"
      title="Input"
    >
      <input
        className="css-1kkc718"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Placeholder"
        type="password"
        value="qwertyuiop"
      />
      <div
        className="css-jgwfcl"
      >
        <button
          className="css-pa2svn"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="css-1v31moo"
            viewBox="0 0 16 16"
          >
            <use
              xlinkHref="#icon-eye-opened"
            />
          </svg>
        </button>
      </div>
    </div>
    <p
      className="css-5ut22t"
    />
  </label>,
  <label
    className="css-kf3i1p"
  >
    <div
      className="css-13kfgn1"
    >
      <span
        className="css-dpor9z"
      >
        Label
      </span>
      <span
        className="css-1dhlryp"
      >
        Sub title
      </span>
    </div>
    <div
      className="css-1prx3ty"
      title="Input"
    >
      <input
        className="css-1ssefzq"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Placeholder"
        value="qwertyuiop"
      />
    </div>
    <p
      className="css-5ut22t"
    >
      Field is required
    </p>
  </label>,
  <label
    className="css-kf3i1p"
  >
    <div
      className="css-13kfgn1"
    >
      <span
        className="css-dpor9z"
      >
        Multiline labeled input
      </span>
      <span
        className="css-1dhlryp"
      >
        Sub title
      </span>
    </div>
    <div
      className="css-1izkox9"
      title="Input"
    >
      <textarea
        className="css-1yq8nnk"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Placeholder"
        rows={4}
        value="qwertyuiop"
      />
    </div>
    <p
      className="css-5ut22t"
    >
      Field is required
    </p>
  </label>,
]
`;
